File: python/matrices/TercerEjercicio.py
matriz = [
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9]
]

print("matriz original:")

for i in range(len(matriz)):
    for j in range(len(matriz[i])):
        print(matriz[i][j], end=" ")
    print()

suma = 0

for i in range(len(matriz)):
    for j in range(len(matriz[i])):
        suma = suma + matriz[i][j]

print("suma de todos los elementos:", suma)

matriz[0], matriz[len(matriz) - 1] = matriz[len(matriz) - 1], matriz[0]

# Imprimir matriz después del intercambio
print("matriz después de intercambiar la primera fila con la última:")

for i in range(len(matriz)):
    for j in range(len(matriz[i])):
        print(matriz[i][j], end=" ")
    print()

File: C#/matrices/TercerEjercicio.cs
using System;

class TercerEjercicio
{
    static void Main()
    {
        int[,] matriz =
        {
            { 1, 2, 3 },
            { 4, 5, 6 },
            { 7, 8, 9 }
        };

        Console.WriteLine("Matriz original:");

        for (byte i = 0; i < matriz.GetLength(0); i++)
        {
            for (byte j = 0; j < matriz.GetLength(1); j++)
            {
                Console.Write(matriz[i, j] + "\t");
            }

            Console.WriteLine();
        }
        int suma = 0;

        for (byte i = 0; i < matriz.GetLength(0); i++)
        {
            for (byte j = 0; j < matriz.GetLength(1); j++)
            {
                suma += matriz[i, j];
            }
        }

        Console.WriteLine("La suma de los números de la matriz es: " + suma);
        int temporal;

        for (byte j = 0; j < matriz.GetLength(1); j++)
        {
            temporal = matriz[0, j];
            matriz[0, j] = matriz[matriz.GetLength(0) - 1, j];
            matriz[matriz.GetLength(0) - 1, j] = temporal;
        }

        Console.WriteLine("\nMatriz después de intercambiar primera y última fila:");

        for (byte i = 0; i < matriz.GetLength(0); i++)
        {
            for (byte j = 0; j < matriz.GetLength(1); j++)
            {
                Console.Write(matriz[i, j] + "\t");
            }

            Console.WriteLine();
        }
    }
}